rpc/src/server/mod.rs

```diff
@@ -47,6 +50,19 @@ pub type RpcServiceEnvironmentRef = Arc<RpcServiceEnvironment>;
 pub struct RpcCollectedState {
     #[get = "pub(crate)"]
     current_head: Arc<BlockHeaderWithHash>,
+
+    // Wakers for open streams (monitors) that access the mempool state
+    streams: HashMap<Uuid, Waker>,
```

Comment:
@adonagy 
I would prefer to spread that hacked `uuid` as less as possible, so 
I would better clean this little:
- move `streaming_state` to `shell-integration`
- create alias type in `streaming_state`, e.g.
```
pub type StreamWakers = HashMap<Uuid, Waker>;
```
- remove `uuid` from `rpc/Cargo.toml` and `shell/Cargo.toml`
- add that hacked `uuid` just to `shell-integration/Cargo.toml`

Reply:
anyway, if this will be fixed: https://github.com/serde-rs/serde/issues/2096
we could return to the normal `uuid` dependency

Reply:
@adonagy 
and also if we want to add here that hacked `uuid`, please change it everywhere, I see it in `monitoring` module,
that we use just the same `uuid` everywhere